cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc

```diff
@@ -66,18 +65,20 @@ class TestUnaryArithmetic : public TestBase {
     return *arrow::MakeScalar(type_singleton(), value);
   }
 
+  void SetUp() override {}
+
   // (CScalar, CScalar)
   void AssertUnaryOp(UnaryFunction func, CType argument, CType expected) {
     auto arg = MakeScalar(argument);
     auto exp = MakeScalar(expected);
-    ASSERT_OK_AND_ASSIGN(auto actual, func(arg, options_, nullptr));
+    ASSERT_OK_AND_ASSIGN(auto actual, func(arg, options_, NULLPTR));
```

Comment:
Those changes shouldn't be necessary, `NULLPTR` is only required in header files.